packages/flutter_web_plugins/lib/src/navigation/utils.dart

```diff
@@ -15,30 +13,30 @@ AnchorElement _urlParsingNode;
 String extractPathname(String url) {
   // TODO(mdebbar): Use the `URI` class instead?
   _urlParsingNode ??= AnchorElement();
-  _urlParsingNode.href = url;
-  final String pathname = _urlParsingNode.pathname;
+  _urlParsingNode!.href = url;
+  final String pathname = _urlParsingNode!.pathname ?? '';
   return (pathname.isEmpty || pathname[0] == '/') ? pathname : '/$pathname';
 }
 
-Element _baseElement;
+Element? _baseElement;
 
 /// Finds the <base> element in the document and returns its `href` attribute.
 ///
 /// Returns null if the element isn't found.
-String getBaseElementHrefFromDom() {
+String? getBaseElementHrefFromDom() {
   if (_baseElement == null) {
     _baseElement = document.querySelector('base');
     if (_baseElement == null) {
       return null;
     }
   }
-  return _baseElement.getAttribute('href');
+  return _baseElement!.getAttribute('href');
```

Comment:
Making `_baseElement` `final` would simplify this method greatly:

```dart
return _baseElement?.getAttribute('href');
```

Reply:
Yes, it becomes so simple that it can be defined with a fat arrow and still fit in the line width :) Done that.